cmd/geth/flag.go

```diff
@@ -453,6 +455,81 @@ func makeNodeName(version string, ctx *cli.Context) string {
 	return name
 }
 
+func mustMakeMLogDir(ctx *cli.Context) string {
+	if ctx.GlobalIsSet(MLogDirFlag.Name) {
+		p := ctx.GlobalString(MLogDirFlag.Name)
+		if p == "" {
+			glog.Fatalf("Flag %v requires a non-empty argument", MLogDirFlag.Name)
+			return ""
+		}
+		cwd, e := os.Getwd()
+		if e != nil {
+			cwd = "."
+		}
+		rp := common.EnsurePathAbsoluteOrRelativeTo(cwd, p)
+		if filepath.IsAbs(rp) {
+			return rp
+		}
+		ap, e := filepath.Abs(rp)
```

Comment:
Is this line reachable?

From my reading of [getwd](https://golang.org/src/os/getwd.go), it seems always return an absolute path. `EnsurePathAbsoluteOrRelativeTo` will already join `cwd` with `p` if `p` is relative. `filepath.IsAbs` will still then, in the case of a relative path (if that ever happens), joins `rp` again with the current folder. Is this the desired behaviour?

Reply:
If `os.Getwd` returns an error then `cwd = "."`, and `EnsurePathAbsoluteOrRelativeTo` will set `rp := "./mlogs"`

Reply:
Is there a particular reason to use `EnsurePathAbsoluteOrRelativeTo` in this place? Seems that `filepath.Abs(p)` just does the job. Quoted from the docs:

> If the path is not absolute it will be joined with the current working directory to turn it into an absolute path.